docs/10-stories.md

```diff
@@ -0,0 +1,168 @@
+# Stories
+
+Source uses [storybook](https://storybook.js.org/) for its API documentation as well as for visual regression testing.
+
+The Source storybook can be run locally using `yarn storybook`
+
+## Story Configuration
+
+1. Story files should either be names `stories.tsx` or `MyComponent.stories.tsx`
+1. Story titles should follow the pattern `Source/${package}/${component}` e.g. `Source/src-radio/RadioGroup`
+1. Each component should have a Demo story which is designed for documentation purposes and a series of stories detailing specific cases for visual regression testing
+1. Stories should have PascalCase names
+1. The `args` property should be passed to all stories so that the source code is displayed on the docs page
+1. Props should all be commented to autofill the args table
+1. Components should all have doc strings to auto fill the docs page (see [example](#example) below)
+1. Controls should be disabled for all stories except the Demo story
```

Comment:
this is an interesting one. you can use the JSDoc `@ignore` comment to hide controls (example in the [accordion](https://github.com/guardian/source/blob/31af12ae2574addae2e75bd349c92f7e51a0f545/src/core/components/accordion/AccordionRow.tsx#L38), which also gives you a place to add comments explaining why. 

i think i prefer this, but is it doing the same thing?

Reply:
Disabling the controls for the non-demo stories removes the whole controls tab from those stories. For those stories at the moment, we're hard coding the args we want, rather than using `Story.args = {...}`. This means that the controls don't actually work for any of the hard coded values. I opted for removing the controls tab as it's confusing to have non-working controls. An alternative would be to use `Story.args = {...}` everywhere so that the controls did work. I didn't go for this as it was more effort and these stories are predominantly for visual regression testing but happy to go with this approach instead.

The following point is disabling an individual control, similar to the example for Accordion. The difference is that it still displays in the arg table, just without a control. I was doing that for props like `cssOverrides` on the basis that it's useful to show that this prop is available but the control won't do anything in storybook so it's better to remove it. For the AccordionRow example, I'd agree that removing it from the args table completely is the best option, as we don't really want to document this as a prop that a user should pass directly.

Reply:
https://github.com/guardian/source/pull/903/commits/1c9d41642d8850dd1d6e252e2a3a6929b47126c5 adds a point about removing args completely using `@ignore` if they shouldn't be shown in docs

Reply:
👍